### DISCUSSION TITLE
Widget similar to ttk.messagebox

### DISCUSSION THREAD

**Discussion by kidlat14** [TomSchimansky/CustomTkinter#1711] [Q&A] [answered]
Good day,

I have just stumbled upon CustomTkinter and I'm having a blast with it. I have not fully read the documentation so this question is a bit stupid, but does CustomTkinter have a widget similar to the ttk.messagebox from tkinter? 

Thanks.

**Comment by avalon60** [comment, marked as answer]
> Good day,
> 
> I have just stumbled upon CustomTkinter and I'm having a blast with it. I have not fully read the documentation so this question is a bit stupid, but does CustomTkinter have a widget similar to the ttk.messagebox from tkinter?
> 
> Thanks.


It's not part of the official CustomTkinter package, but there is a really good one (which I use): CTkMessagebox.

`from CTkMessagebox import CTkMessagebox`

It's published to PyPi so you can just pull it with pip (or through your IDE).

See: https://github.com/Akascape/CTkMessagebox

**Reply by kidlat14** [reply]
Thanks. This should work.

**Reply by avalon60** [reply]
@kidlat14 by the way, I would recommend using the "options" parameter like so:

```
        if self._json_state == 'dirty':
            confirm = CTkMessagebox(master=self.ctk_control_panel,
                                    title='Confirm Action',
                                    message=f'You have unsaved changes. Are you sure you wish to switch themes?',
                                    options=["Yes", "No"])
            response = confirm.get()
            if response == 'No':
                return
```

When you do it this way, you don't use option_1, option_2...

**Reply by kidlat14** [reply]
Thanks. This is very helpful.

**Comment by Akascape** [comment]
@kidlat14 Try Ctkmessagebox created by me.

You can find more interesting packages here:
https://github.com/Akascape/tkinter-toolkit
